app/models/work.rb

```diff
@@ -310,7 +310,12 @@ def self.find_by_url(url)
     url = UrlFormatter.new(url)
     Work.where(:imported_from_url => url.original).first ||
       Work.where(:imported_from_url => [url.minimal, url.no_www, url.with_www, url.encoded, url.decoded]).first ||
-      Work.where("imported_from_url LIKE ? OR imported_from_url LIKE ?", "%#{url.encoded}%", "%#{url.decoded}%").first
+      Work.find do |w|
```

Comment:
This doesn't seem like something we can do in production - it would get every single work from the database and load them into memory.